primitives/ec/wif_test.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2013, 2014 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package primitives
+
+import (
+	"encoding/hex"
+	"testing"
+
+	chaincfg "github.com/bitcoin-sv/go-sdk/transaction/chaincfg"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEncodeDecodeWIF(t *testing.T) {
+	mainWif := "Kxfd8ABTYZHBH3y1jToJ2AUJTMVbsNaqQsrkpo9gnnc1JXfBH8mn"
+	testWif := "cP2cb5BJycySSVSH7scRPUyN5ao1XpgXUv1DwDcCHuG1ZGkJQxzH"
+	privHex := "2b2afb5ea8d8c623acd6744547628988d86787003bb970afe15d471b727db79c"
+
+	mainPriv, _ := PrivateKeyFromWif(mainWif)
+	testPriv, _ := PrivateKeyFromWif(testWif)
+	assert.Equal(t, mainWif, mainPriv.Wif(&chaincfg.MainNet))
```

Comment:
This chaincfg.MainNet is a bit ugly. I recall thinking this stuff mattered when first using bsvjs - having the whole environment get set to a specific chain. Now I know it's literally just a prefix if you want to see it as a string, it feels like we can tuck it away somewhere.